src/main/java/com/systemmeltdown/meltdownlog/topics/IntegerTopic.java

```diff
@@ -0,0 +1,41 @@
+package com.systemmeltdown.meltdownlog.topics;
+
+import com.systemmeltdown.meltdownlog.lib.Utils;
+
+/**
+ * Logs integers as a data topic.
+ */
+public class IntegerTopic extends LogTopic {
+	private final int m_roundingFactor;
+	private int m_lastRoundedValue = Integer.MIN_VALUE;
+	private long m_lastNanos = Long.MIN_VALUE;
+
+	public IntegerTopic(final String name, final int roundingFactor) {
+		super(name, Integer.class);
+		m_roundingFactor = roundingFactor;
+	}
+
+	public void log(final int value) {
+		log(value, System.nanoTime());
+	}
+
+	public void log(final int value, final long nanos) {
+		final int roundedValue = Utils.roundByFactor(value, m_roundingFactor);
+		if (roundedValue == m_lastRoundedValue) {
```

Comment:
Same stuff here

Reply:
I have now consolidated this functionality in a single class, DataTopic.
Thanks!